### PR TITLE
Replace kubectl wait command with RayClusterAddCREvent

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?
#704 causes `kubectl wait` to be flaky, and thus I used `RayClusterAddCREvent` to reduce flakiness.

In #704, `kubectl wait` sometimes waits for 1 head pod and 2 worker pods (goal state is 1 head pod and 1 worker pod) due to the inconsistency between Kubernetes API server and informer cache, and thus the function `create_kuberay_cluster` is flaky. `RayClusterAddCREvent` specifies the goal state (1 head pod and 1 worker pod) directly to reduce flakiness.

https://github.com/ray-project/kuberay/blob/17264a6f409208ee24cbb743c59fa40c9eed856a/tests/framework/prototype.py#L204-L231

## Related issue number
https://github.com/ray-project/kuberay/issues/704

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: tests/kuberay_utils/utils.py [hunk 1]
```diff
@@ -152,12 +159,16 @@ def delete_cluster():
 
 
 def download_images(images):
+    """Pull images from DokcerHub if do not exist."""
```
Comment (DmitriGekhtman) [inline tests/kuberay_utils/utils.py:162]: Dokcer->Docker

File: tests/kuberay_utils/utils.py [hunk 2]
```diff
@@ -80,37 +87,37 @@ def apply_kuberay_resources(images, kuberay_operator_image, kuberay_apiserver_im
...
-    rtn = shell_run(
-        'kubectl wait --for=condition=ready pod -l rayCluster=raycluster-compatibility-test --all --timeout=900s')
-    shell_run('kubectl get pods -A')
-    if rtn != 0:
+    try:
+        # Deploy a NodePort service to expose ports for users.
+        shell_assert_success(f'kubectl apply -f {raycluster_service_file}')
+        # Create a RayCluster
+        ray_cluster_add_event = RayClusterAddCREvent(
+            context['cr'], [], 90, namespace='default', filepath = context['filepath'])
```
Comment (DmitriGekhtman) [inline tests/kuberay_utils/utils.py:111]: Would you mind naming all of the args?

**Review by DmitriGekhtman — COMMENTED** [review]
Looks good. Could you explain in the issue description how this solves the issue?

File: tests/kuberay_utils/utils.py (hunk 1 above)
Comment (kevin85421) [reply tests/kuberay_utils/utils.py]: Updated.

File: tests/kuberay_utils/utils.py (hunk 2 above)
Comment (kevin85421) [reply tests/kuberay_utils/utils.py]: Updated.

**Comment by kevin85421** [comment]
> Looks good. Could you explain in the issue description how this solves the issue?

Thank @DmitriGekhtman for the review! I have updated the PR description.

**Review by DmitriGekhtman — APPROVED** [review]
Nice, great that you were able to use a component of the test framework.

**Pull request merged**